src/sentry/tasks/post_process.py

```diff
@@ -204,109 +205,87 @@ def should_issue_owners_ratelimit(project_id: int, group_id: int, organization_i
+            assignees_exists,
+            (ASSIGNEE_EXISTS_DURATION if assignees_exists else ASSIGNEE_DOES_NOT_EXIST_DURATION),
+        )
 
-                with sentry_sdk.start_span(
-                    op="post_process.handle_owner_assignment.debounce_issue_owners"
-                ):
-                    issue_owners_key = ISSUE_OWNERS_DEBOUNCE_KEY(group.id)
-                    debounce_issue_owners = cache.get(issue_owners_key)
+    if assignees_exists:
+        metrics.incr("sentry.task.post_process.handle_owner_assignment.assignee_exists")
+        return
 
-                    if debounce_issue_owners:
-                        metrics.incr("sentry.tasks.post_process.handle_owner_assignment.debounce")
-                        return
+    issue_owners_key = ISSUE_OWNERS_DEBOUNCE_KEY(group.id)
+    debounce_issue_owners = cache.get(issue_owners_key)
 
-                with metrics.timer("post_process.process_owner_assignments.duration"):
-                    with sentry_sdk.start_span(
-                        op="post_process.handle_owner_assignment.get_issue_owners"
-                    ):
-                        if killswitch_matches_context(
-                            "post_process.get-autoassign-owners",
-                            {
-                                "project_id": project.id,
-                            },
-                        ):
-                            # see ProjectOwnership.get_issue_owners
-                            issue_owners: Sequence[tuple[Rule, Sequence[Team | RpcUser], str]] = []
-                        else:
-                            issue_owners = ProjectOwnership.get_issue_owners(project.id, event.data)
-
-                            # Cache for 1 day after we calculated. We don't need to move that fast.
-                            cache.set(
-                                issue_owners_key,
-                                True,
-                                ISSUE_OWNERS_DEBOUNCE_DURATION,
-                            )
+    if debounce_issue_owners:
+        metrics.incr("sentry.tasks.post_process.handle_owner_assignment.debounce")
+        return
 
-                    with sentry_sdk.start_span(
-                        op="post_process.handle_owner_assignment.handle_group_owners"
-                    ):
-                        if issue_owners:
-                            try:
-                                handle_group_owners(project, group, issue_owners)
-                            except Exception:
-                                logger.exception("Failed to store group owners")
-                        else:
-                            handle_invalid_group_owners(group)
+    if killswitch_matches_context(
+        "post_process.get-autoassign-owners",
+        {
+            "project_id": project.id,
+        },
+    ):
+        # see ProjectOwnership.get_issue_owners
+        issue_owners: Sequence[tuple[Rule, Sequence[Team | RpcUser], str]] = []
```

Comment:
do we need to go through `handle_invalid_group_owners` in this branch? the old code without the early-return would eventually call `handle_invalid_group_owners(group)`.

Reply:
although imo if the killswitch is set, returning without doing anything seems like the right call

Reply:
yeah, this is a fair point. it is a behavior change. i'll modify the function to call `handle_invalid_group_owners`. i feel suspicious about even hitting that if the killswitch is activated, but will keep the same.

Reply:
went ahead and changed that else back